File: .claude/hooks/notify-slack.mjs
#!/usr/bin/env node
// @hookstack notification-slack
import { execSync } from "node:child_process";
// @hookstack notification-slack
// Envoie une notification Slack quand Claude veut notifier l'utilisateur (Notification)
import { readFileSync } from "node:fs";
import { fileURLToPath } from "node:url";

function defaultExec(cmd) {
	execSync(cmd, { stdio: "ignore", timeout: 10_000 });
}

export function run(
	input,
	{
		exec = defaultExec,
		webhook = process.env.SLACK_WEBHOOK_URL ?? "",
		projectDir = process.env.CLAUDE_PROJECT_DIR,
	} = {},
) {
	if (!webhook) return null;

	const message = input.message ?? input.notification ?? "";
	if (!message) return null;

	const project = projectDir?.split("/").pop() ?? "Claude";
	const payload = JSON.stringify({ text: `*[${project}]* ${message}` });

	try {
		exec(
			`curl -s -X POST -H 'Content-type: application/json' --data '${payload.replace(/'/g, "'\\''")}' '${webhook}'`,
		);
	} catch {
		// Échec réseau — non bloquant
	}
	return payload;
}

/* v8 ignore next 4 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
	const input = JSON.parse(readFileSync(0, "utf8"));
	run(input);
}

File: .claude/hooks/notify-slack.py
#!/usr/bin/env python3
# @hookstack notification-slack
"""Sends a Slack notification when Claude wants to notify the user (Notification)."""
import json
import os
import subprocess
import sys


def _default_exec(cmd):
    subprocess.run(cmd, shell=True, capture_output=True, timeout=10)


def run(
    input_data=None,
    *,
    exec_cmd=None,
    webhook=None,
    project_dir=None,
):
    if exec_cmd is None:
        exec_cmd = _default_exec
    if webhook is None:
        webhook = os.environ.get("SLACK_WEBHOOK_URL") or ""
    if project_dir is None:
        project_dir = os.environ.get("CLAUDE_PROJECT_DIR")

    if not webhook:
        return None

    data = input_data or {}
    message = data.get("message") or data.get("notification") or ""
    if not message:
        return None

    project = project_dir.split("/")[-1] if project_dir else "Claude"
    payload = json.dumps({"text": f"*[{project}]* {message}"})

    try:
        exec_cmd(
            f"curl -s -X POST -H 'Content-type: application/json' --data '{payload.replace(chr(39), chr(39) + chr(92) + chr(39) + chr(39))}' '{webhook}'"
        )
    except Exception:
        # Échec réseau — non bloquant
        pass
    return payload


if __name__ == "__main__":
    data = json.load(sys.stdin)
    run(data)
